library/std/src/net/endian.rs

```diff
@@ -0,0 +1,37 @@
+//! Define internal endianness primitives for networking.
+
+/// A Big Endian u32.
+///
+/// Mostly used as inner type of IpvAddr.
+#[repr(transparent)]
+#[derive(Copy, Clone, PartialEq)]
+pub(super) struct BeU32(u32);
+
+impl BeU32 {
+    /// The argument `n` is interpreted as big endian.
+    ///
+    /// Caller has to ensure this invariant are met.
+    pub const fn from_be(n: u32) -> Self {
+        Self(n)
+    }
+
+    /// Interprets `arr` in big endian order.
+    pub const fn from_bytes(arr: [u8; 4]) -> Self {
+        // The bytes are already in Big Endian order. So native conversion it as `u32`
+        // will make it an BE `u32` regardless of target endian.
+        Self(u32::from_ne_bytes(arr))
+    }
+
+    /// Returns a big endian u32.
+    pub const fn to_be(&self) -> u32 {
+        // Already as big endian.
+        self.0
+    }
+
+    /// Returns a big endian array.
+    pub const fn to_bytes(&self) -> [u8; 4] {
```

Comment:
Maybe add a comment saying that the return value is in native endianess? Or maybe even make that part of the method name?

Reply:
Oh wait it doesn't return native endianess but big-endian? This is confusing.

Reply:
Yeah, I was confusing too. Because self is big endian already. We just need the `to_ne_bytes` method to turn it in to big endian array.
For example: the value `0xFFEE` stores as `0xFF 0xEE` in big endian order in memory, regardless of target endian.

Reply:
It might be less confusing to do pointer casts here that to try to re-use the existing methods.

Reply:
Pointer casts or Union fields casts involve unsafe though.

Reply:
I know. I still think it would be clearer.

Reply:
I reword doc a little bit. Hope you find that less confusing.